script

```diff
@@ -693,52 +693,40 @@ only_send_error_notifications="0"
-
-              # run compare function. compare will not run if compare_files is disabled.
-              run_compare "$disk" "$backup_location/$vm/$new_disk" "vdisk"
-
-            elif [ "$inline_zstd_compress" -eq 1 ]; then
-              copy_file "$disk" "$backup_location/$vm/$timestamp$new_disk.zst" "$rsync_dry_run_option" "inline_zstd_compress" "$rsync_only"
+            local dest_disk="$backup_location/$vm/$new_disk"
 
-              # make sure copy has current date/time for modified attribute so that removing old backups by date will work.
-              touch -d "now" "$backup_location/$vm/$timestamp$new_disk.zst"
+            if [ "$timestamp_files" -eq 1 ] || [ "$number_of_backups_to_keep" -ne 1 ]; then
+              # if timestamps are enabled or more than one backup is being kept, add a timestamp to the destination filename.
+              dest_disk="$backup_location/$vm/$timestamp$new_disk"
+            fi
 
+            # copy or pretend to copy the vdisk to the backup location specified by the user.
+            if [ "$inline_zstd_compress" -eq 1 ]; then
+              # if inline_zstd_compress is enabled, add ".zst" to the destination filename, and don't run the comparison.
+              dest_disk+=".zst"
+              copy_file "$disk" "$dest_disk" "$rsync_dry_run_option" "inline_zstd_compress" "$rsync_only"
             else
 
-              # check to see if an existing backup was already copied. if so, use rsync. if not, use cp.
-              if [ ! -f "$backup_location/$vm/$timestamp$new_disk" ]; then
-                copy_file "$disk" "$backup_location/$vm/$timestamp$new_disk" "$rsync_dry_run_option" "sparse" "$rsync_only"
-
-                # make sure copy has current date/time for modified attribute so that removing old backups by date will work.
-                touch -d "now" "$backup_location/$vm/$timestamp$new_disk"
-
+              if [ -f "$disk_dest" ]; then
+                # if there is an existing backup, use rsync to just copy the changes (i.e. delta_sync).
+                copy_file "$disk" "$dest_disk" "$rsync_dry_run_option" "inplace" "$rsync_only"
               else
-
-                # use rsync because delta_sync was enabled.
-                copy_file "$disk" "$backup_location/$vm/$timestamp$new_disk" "$rsync_dry_run_option" "inplace" "$rsync_only"
-
-                # make sure copy has current date/time for modified attribute so that removing old backups by date will work.
-                touch -d "now" "$backup_location/$vm/$timestamp$new_disk"
+                # otherwise, copy the whole file (with sparse support)
+                copy_file "$disk" "$dest_disk" "$rsync_dry_run_option" "sparse" "$rsync_only"
               fi
 
               # run compare function. compare will not run if compare_files is disabled.
-              run_compare "$disk" "$backup_location/$vm/$timestamp$new_disk" "vdisk"
+              run_compare "$disk" "$disk_dest" "vdisk"
```

Comment:
this line references $disk_dest, but I don't see it assigned anywhere? Possibly this is a typo and should be $dest_disk instead. this also occurs on line 710

Reply:
typo, fixed.